targets/psa_switch/psa_switch.cpp

```diff
@@ -337,17 +420,57 @@ PsaSwitch::ingress_thread() {
     ingress_mau->apply(packet.get());
     packet->reset_exit();
 
-    // prioritize dropping if marked as such - do not move below other checks
-    const auto &f_drop = phv->get_field("psa_ingress_output_metadata.drop");
-    if (f_drop.get_int()) {
+    const auto &f_ig_cos = phv->get_field("psa_ingress_output_metadata.class_of_service");
+    const auto ig_cos = f_ig_cos.get_uint();
+
+    // ingress cloning - sent in their pre-ingress-thread state to either an egress
+    //                   port or a multicast group
+    //                 - dropped packets should still be cloned - do not move below drop
+    unsigned int clone = phv->get_field("psa_ingress_output_metadata.clone").get_uint();
+    if (clone) {
+      MirroringSessionConfig config;
+      unsigned int clone_session_id = phv->get_field("psa_ingress_output_metadata.clone_session_id").get_uint();
+      bool is_session_configured = mirroring_get_session(
+          static_cast<mirror_id_t>(clone_session_id), &config);
+      BMLOG_DEBUG_PKT(*packet, "Cloning packet at ingress to session id {}", clone_session_id);
```

Comment:
is having this log message confusing in case the session is not configured? I know that I didn't do it for simple_switch, but maybe a log message to indicate that a session id which hasn't been configured has been requested would be a good idea. I wouldn't make it an error or even a warning, but it could be helpful when debugging. @jafingerhut what do you think?

Reply:
I am certainly all for easier debugging of P4 code from the contents of the bmv2 log, so having different log messages for a configured vs. not-configured session could be nice.  Perhaps have the above log message inside the 'if (is_session_configured) {' 'then' branch below, and a slightly different one in an 'else' branch that said something like "Cloning packet at ingress to unconfigured session id {} causes no clone packets to be created" ?

Reply:
+1 for andy's suggestion. Adding in next commit.